lib/src/Base/Algo/CubaIntegration.cxx

```diff
@@ -0,0 +1,311 @@
+  }
+  /* Cuba computes an integral over the unit hypercube, one has to multiply its
+     output by the volume of the true integration interval. */
+  const Point ptOut(intervalVolume * (*pFunction)(ptIn));
+  if (ptOut.getSize() != static_cast<UnsignedInteger>(*ncomp))
+  {
+    throw InvalidArgumentException(HERE) << "Error: expected number of components and dimension of the function output do not match, here expected number of components= " << *ncomp << " and dimension of the function output=" << ptOut.getSize();
+  }
+  std::copy(ptOut.begin(), ptOut.end(), f);
+
+  return 0;
+}
+
+/* Compute an approximation of \int_{I}f(x)dx, where I
+ * is an multidimensional interval and f a multidimensional function
+ */
+Point CubaIntegration::integrate(const Function & function,
+                                 const Interval & interval) const
+{
+#ifdef OPENTURNS_HAVE_CUBA
+  if (function.getInputDimension() != interval.getDimension()) throw InvalidArgumentException(HERE) << "Error: interval dimension and input dimension have to match, here interval dimension="<< interval.getDimension() << " and input dimension=" << function.getInputDimension();
+  const UnsignedInteger inputDimension = interval.getDimension();
+  const UnsignedInteger outputDimension = function.getOutputDimension();
+  /*
+   * For the number of regions, of evaluations, and the failure indicator, pointers
+   * to int, not long (aka SignedInteger) are expected by Cuba.
+   */
+  int nregions;
+  int neval;
+  int fail;
+  Point integral(outputDimension);
+  Point error(outputDimension);
+  Point prob(outputDimension);
+
+  /* Providing the integrator with pointers to the function and to the integration interval */
+  void* ppFunctionInterval[2];
+  ppFunctionInterval[0] = (void*)(&function);
+  ppFunctionInterval[1] = (void*)(&interval);
+
+  /* Setting constant parameters common to all routines */
+  const UnsignedInteger nvec = 1; /* Maximum number of points to be given to the integrand at each invocation */
+  const UnsignedInteger mineval = 0; /* The minimum number of integrand evaluations required */
+  const SignedInteger seed = 0; /* The seed for the pseudo-random-number generator */
+
+  /* Calling the chosen optimization routine, with specific parameters for each of them */
+  if (optRoutine_ == "cuhre")
+  {
+    /* Cuhre-specific parameters */
+    const UnsignedInteger key = 0; /* Default integration rule */
+
+    Cuhre(inputDimension, outputDimension, computeIntegrand,
+            (void*)(ppFunctionInterval), nvec, epsRel_, epsAbs_, flags_,
+            mineval, maxeval_, key, NULL, NULL, &nregions, &neval, &fail,
+            const_cast<double*>(integral.data()),
+            const_cast<double*>(error.data()),
+            const_cast<double*>(prob.data()));
+  }
+  else if (optRoutine_ == "divonne")
+  {
+    /* Divonne-specific parameters */
+    const SignedInteger key1 = 47; /* Sampling parameter for the partitioning phase */
```

Comment:
I would suggest to introduce the corresponding ResourceMap keys, so that we can set these parameters at will, without necessarily providing dedicated methods.